src/web/components/TeamMember/TeamMembersTable.spec.tsx

```diff
@@ -0,0 +1,107 @@
+  );
+};
+
+describe('manage team member functionality testing', () => {
+  let participantContextValue: ParticipantWithSetter;
+  let userContextValue: UserContextWithSetter;
+  let mockParticipant: ParticipantDTO;
+  let mockUser: UserWithParticipantRoles;
+  beforeAll(() => {
+    mockParticipant = createMockParticipant();
+    participantContextValue = createParticipantContextValue(mockParticipant);
+    mockUser = createMockUser([mockParticipant] as Participant[]);
+  });
+
+  const teamMemberManageRoles = [
+    { id: UserRoleId.UID2Support, roleName: 'UID2Support' },
+    { id: UserRoleId.Admin, roleName: 'Admin' },
+  ];
+  it.each(teamMemberManageRoles)(
+    'should include all manage team member functionality',
+    (manageRole) => {
+      // setting uid2 support to false to make sure the roles are tested
+      mockUser.isUid2Support = false;
+      mockUser.currentParticipantUserRoles = [manageRole];
+      mockParticipant.currentUserRoleIds = [manageRole.id];
+      userContextValue = createUserContextValue(mockUser);
+      renderTeamMembersTableWithContext(userContextValue, participantContextValue, [mockUser]);
+      const addTeamMemberButton = screen.getByRole('button', { name: 'Add Team Member' });
+      expect(addTeamMemberButton).toBeInTheDocument();
+      const actionsHeader = screen.getByRole('columnheader', { name: 'Actions' });
+      expect(actionsHeader).toBeInTheDocument();
+      expect(screen.getByLabelText('Edit Team Member')).toBeInTheDocument();
+    }
+  );
+
+  it('should not include any manage team member functionality', () => {
+    mockUser.isUid2Support = false;
+    mockUser.currentParticipantUserRoles = [{ id: UserRoleId.Operations, roleName: 'Operations' }];
+    mockParticipant.currentUserRoleIds = [UserRoleId.Operations];
+    userContextValue = createUserContextValue(mockUser);
+    renderTeamMembersTableWithContext(userContextValue, participantContextValue, [mockUser]);
+    expect(screen.queryByRole('button', { name: 'Add Team Member' })).not.toBeInTheDocument();
+    expect(screen.queryByRole('columnheader', { name: 'Actions' })).not.toBeInTheDocument();
+    expect(screen.queryByLabelText('Edit Team Member')).not.toBeInTheDocument();
+  });
+
+  it('should include all manage team member functionality when user has uid2support', () => {
+    mockUser.isUid2Support = true;
+    userContextValue = createUserContextValue(mockUser);
+    renderTeamMembersTableWithContext(userContextValue, participantContextValue, [mockUser]);
+    const addTeamMemberButton = screen.getByRole('button', { name: 'Add Team Member' });
+    expect(addTeamMemberButton).toBeInTheDocument();
+    const actionsHeader = screen.getByRole('columnheader', { name: 'Actions' });
+    expect(actionsHeader).toBeInTheDocument();
+    expect(screen.getByLabelText('Edit Team Member')).toBeInTheDocument();
+  });
+
+  it('should not include any manage team member functionality when user does not have uid2support', () => {
```

Comment:
nit: do we strictly need this test? it seems essentially the same as `should not include any manage team member functionality`

Reply:
It is the same expected result, but since we're not really checking the support role within the roles array, I think it's still useful to test the direct `user.IsUid2Support` field.  If anything, we probably don't need to one above to also check UserRoleId.Uid2Support

Reply:
Actually - I have to take out that check, or the test will fail with the change to `isUserAdminOrSupport` - so now I've updated the test above to just test with the Admin role